doc/api/fs.md

```diff
@@ -70,6 +70,76 @@ the entire process until they complete--halting all connections.
 The relative path to a filename can be used. Remember, however, that this path
 will be relative to `process.cwd()`.
 
+For most `fs` module functions, the `path` or `filename` argument may be passed
+as a WHATWG [URL][] object. Only [URL][] objects using the `file:` protocol are
+supported. Use of [URL][] objects is currently still experimental.
+
+```js
+const fs  = require('fs');
+const { URL } = require('url');
+const file_url = new URL('file:///tmp/hello');
+
+fs.readFile(file_url, 'utf-8', (err,data) => {
```

Comment:
IMO it would be easier to use the synchronous API (`fs.readFileSync`) instead. You can even leave off the `encoding` argument since it's not necessary to demonstrate URL support.

```js
fs.readFileSync(fileURL);
```